### DISCUSSION TITLE
Cloudflare SSL handshake failed Error 525

### DISCUSSION THREAD

**Discussion by yisusheng** [misskey-dev/misskey#11918] [Q&A]
Hello everyone, I'm currently deploying the MIsskey project using the official Misskey Docker image. I have successfully deployed it, and I can access it via the IP address. I have also followed the instructions to configure SSL. However, when I try to access it using the HTTPS domain, I receive a "Cloudflare 525 SSL handshake failed" error.
The most peculiar thing is that I was able to access it through the domain name right after a successful deployment. However, after a few minutes, I couldn't access it anymore even after refreshing the page. I have opened port 443, and the certificate configuration for Nginx is passing the checks. I have searched for information for a long time, but I currently don't know how to troubleshoot this issue.

**Comment by spirillen** [comment]
> "Cloudflare 525 SSL handshake failed

That is because CloudFlare are decrypting the connection and Serves you a faked SSL, do to the fact the connection is INSECURE and UNENCRYPTED as CloudFlare is TRUE MITM for collecting all data about you and any other visiting the hacked connection to the domain in question.

Drop the hackers and get a real connection to your server, that way you can gain peoples trust in your server as they can feel them self safe.

**Comment by LyraHarring** [comment]
When you are deploying the Misskey project and trying to access it through the HTTPS domain, then you are receiving the error "[Cloudflare 525 SSL handshake failed](https://certera.com/blog/how-to-fix-ssl-handshake-failed-error/)"  occurs when a secure connection between a server and a client fails to establish. The common errors, incompatible protocols or cipher suites, incorrect server name indication (SNI), Firewall or NAT issues, time synchronization errors, new plugins or extensions that have been installed, and Malware or virus infections, etc. Some fixes for the error are also as follows:-

1) Correct the System Date and Time
2) Update Your Web Browser
3) Check that your browser supports the most recent TLS protocol
4) Check whether the certificate is valid or not
5) Check your server configuration

I hope it helps!
